web/src/stream_data.ts

```diff
@@ -365,32 +365,10 @@ export function get_streams_for_user(user_id: number): {
     };
 }
 
-export function get_invite_stream_data(): InviteStreamData[] {
-    function get_data(sub: StreamSubscription): InviteStreamData {
-        return {
-            name: sub.name,
-            stream_id: sub.stream_id,
-            invite_only: sub.invite_only,
-            is_web_public: sub.is_web_public,
-            default_stream: default_stream_ids.has(sub.stream_id),
-        };
-    }
-
-    const streams = [];
-
-    // Invite users to all default streams...
-    for (const stream_id of default_stream_ids) {
-        const sub = sub_store.get(stream_id)!;
-        streams.push(get_data(sub));
-    }
-
-    // ...plus all your subscribed streams (avoiding repeats).
-    for (const sub of subscribed_subs()) {
-        if (!default_stream_ids.has(sub.stream_id)) {
-            streams.push(get_data(sub));
-        }
-    }
-
+export function get_invite_stream_data(): StreamSubscription[] {
+    const user_id = people.my_current_user_id();
+    const {subscribed, can_subscribe} = get_streams_for_user(user_id);
```

Comment:
I think this will work because we would not call this code if user is not allowed to subscribe others as the widget itself will not be visible, but the logic used to `can_subscribe` considers whether the user can subscribe himself or not and not subscribe others.

So, should be better to just fitler the streams by using `can_subscribe_others` function so that we do not accidentally break it while changing how the permission to subscribe others is computed.